enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,255 @@
+- this can be very confusing, because user running gather B can get result of the gather A (e.g there could be different gatherers enabled, which would produce different results)
+
+Mitigation: This is the very first question in the [Open questions](#open-questions) section, but perhaps reporting the Insights analysis result in one place could slightly mitigate this problem (the status part could tell whether it was a periodic gather or user-triggered gather). 
+Another option would be to disallow the parallel gather execution - if the Insights operator sees a new CR instance (or any other running gather process) then it interrupts any gathering in progress. 
+
+### Drawbacks
+
+- Very frequent data gathering in a cluster can produce a significant load on the processing part in the console.redhat.com and can overwhelm the processing pipeline. 
+Theoretically, we could limit the number of user-trigerred gathering over time, but this problem can happen today when something is deleting the Insights operator pod on a regular basis with appropriate timing (so that data can be uploaded).  
+- When the newly spawned container is failing to upload the data then it can be running for some longer time (couple of hours). 
+
+
+## Design Details
+
+### Open Questions
+- How to solve the problem with only one Insights analysis result available in a cluster? How to avoid the "uncertainty" of the Insights analysis results described in the [Risk and Mitigations](#risks-and-mitigations) section? 
+- Should the newly created gathering process update the clusteroperator conditions (e.g it fails to upload the data, should it update the `Degraded` status as it is today)? This can probably be racy too or it can lead to temporary and inconsistent conditions state. 
+- What all is required to spawn an extra container? Do we need a new image and a new binary (note that the Insights operator currently includes some additional features - such as pulling SCA certificates - that are not required for a new container/process)?
+- Is some clean-up of the old `datagather.insights.openshift.io` CRs (and completed containers/pods) required? I think it would make sense to keep them only for 24h.
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+The plan is to introduce the first version of the new API behind the `TechPreviewNoUpgrade` feature gate.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+The `TechPreviewNoUpgrade` feature gate requirement is removed. 
+
+Other than that we would like to:
+- get some time to gather feedback and some user experience
+- ensure stability
+- do more e2e testing
+
+#### Removing a deprecated feature
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+No special upgrade or downgrade strategy should be needed. The new `datagather.insights.openshift.io` CRD will be introduced with an upgrade to given version (allowing users to run on demand Insights data gathering). This CRD will be removed during the downgrade. 
```

Comment:
Does CVO remove manifests on downgrade?

Reply:
Yeah I think it should remove the CRD definition (as well as all the CRs) or what manifests do you mean pls? I guess I should explicitly mention it, right? Does this represent a new requirement for the CVO?

Reply:
I actually don't think that's a feature CVO has. If it does, great, if it doesn't, you'll need to rethink this, eg, there will be manifests but nothing will observe them on downgrade. Can you please check that the CVO has this feature

Reply:
Downgrade is not supported as I learned from Trevor so I will remove it. He created https://github.com/openshift/enhancements/pull/1290

Reply:
Seems there's some discussion happening about whether ignoring downgrades is right nor not. Typically with a new CRD, the downgrade strategy is "The CRD and CRs will remain in the cluster but no controller will observe them, there should be no ill-effect to the cluster in this scenario", which I think would suffice here.

Pretty sure CVO doesn't do any removal of resources as this currently reads

Reply:
Updated.Thanks